### ISSUE TITLE
Installing goodman using dependencies managers

### ISSUE THREAD

**Issue by gonzalo-bulnes** [snikch/goodman#17]
Hello @snikch and @ddelnano,

**TL;DR**: I think `go get github.com/snikch/goodman/cmd/goodman` would be a better recommendation to install goodman because it behaves in a standard way.

I am getting started with goodman, and I was surprized by the installation instructions:

> `go get github.com/snikch/goodman`
> `go build -o $GOPATH/bin/goodman github.com/snikch/goodman/cmd/goodman`

Specifically, I was surprized by the mandatory `go build`: usually, `go get` does install the binary.

Further on, that additional step turned to be annoying because it doesn't play well with dependencies managers. (I am using [govendor][govendor] but I think the comment is relevant to any dependencies managers which rely on the standard `go get` / `go install` under the hood.)

  [govendor]: https://github.com/kardianos/govendor

What do I mean by "doesn't play well"? If I add goodman to my project dependencies (`govendor fetch github.com/snikch/goodman` in my case), then I expect it to be installed when I install all the dependencies (`govendor install +vendor` in my case). That is particularly true in my build pipeline, where I wish I hadn't to add goodman-specific steps to install everything I need to run my test suite.

It makes sense to me to keep the library at the root level and the command in `cmd/`, I mean I think it's a least a legitimate choice.

But considering that using Dredd hooks requires both the library and the binary to be installed, I wish the default recommendation would take care of that (which it does! ...in a non-standard way).

Using `go get github.com/snikch/goodman/cmd/goodman` (resp. `govendor install github.com/snikch/goodman/cmd/goodman` or whatever you dependencies manager requires) does the trick. And would be a better installation recommendation for the `README`/[wiki][wiki] IMHO.

  [wiki]: https://github.com/snikch/goodman/wiki/Installation

The `main` package depends on the library (`goodman`) which depends on `goodman/transaction`. When requiring `github.com/snikch/goodman/cmd/goodman`, `go get` performs `go fetch` and `go install` for `main` and everything falls in place nicely.

Am I missing something? What you you think? I can open a PR if you want : )

**Comment by ddelnano** [comment]
@gonzalo-bulnes helping rewrite this to be compliant with the Dredd hooks test suite was my first go project so clearly there were plenty of best practice things I was not aware of at the time :)

Since its a really easy fix I can make the change I can make the fix.  The instructions are also in the Dredd repo so I will submit the same change there and link both here.

**Comment by gonzalo-bulnes** [comment]
Awesome @ddelnano ! Thank you : )

**Comment by ddelnano** [comment]
Here is the one for this repo #19

**Comment by ddelnano** [comment]
[Here](https://github.com/apiaryio/dredd/pull/707) is the one on the dredd repo

**Comment by ddelnano** [comment]
I'm gonna wait to merge #19 until the Dredd one is accepted just for the sake of keeping everything consistent but thank you for bringing this up!!

**Comment by gonzalo-bulnes** [comment]
Thank _you_ @ddelnano!

**Comment by ddelnano** [comment]
#19 is merged.
